dnf.spec

```diff
@@ -1,44 +1,78 @@
+# default dependencies
 %global hawkey_version 0.13.3
 %global librepo_version 1.9.0
 %global libcomps_version 0.1.8
 %global rpm_version 4.14.0
+
+# conflicts
+%global conflicts_dnf_plugins_core_version 2.1.3
+
+# override dependencies for rhel 7
 %if 0%{?rhel} == 7
-%global rpm_version 4.11.3-27
+    %global rpm_version 4.11.3-25.el7.centos.1
 %endif
+
+# override dependencies for fedora 26
 %if 0%{?fedora} == 26
-%global rpm_version 4.13.0.1-7
+    %global rpm_version 4.13.0.1-7
+%endif
+
+
+# default build options
+# %%bcond_without sets an --without option -> %%{with foo} is true
+# overriding with %%undefine is not ideal, but it avoids complicated nested conditions
+%bcond_without python2
+%bcond_without python3
+
+# override build options for rhel <= 7
+%if 0%{?rhel} && 0%{?rhel} <= 7
+    %undefine with_python3
```

Comment:
no, don't do this. it breaks `--with/--without` CLI options

Reply:
Could you propose a better, saner option without too many nested if-s, that would also scale easily?

Reply:
```spec
%if 0%{?rhel} && 0%{?rhel} <= 7
%bcond_with python3
%else
%bcond_without python3
%endif
%if 0%{?rhel} >= 8
%bcond_with python2
%else
%bcond_without python2
%endif
```

for instance.